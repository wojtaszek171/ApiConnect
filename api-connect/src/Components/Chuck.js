import React, { Component } from 'react';
import $ from 'jquery';
import 'jquery-ui-dist/jquery-ui';
import ReactHtmlParser from 'react-html-parser';
var loader = require('./ajaxloader.gif');

class Chuck extends Component {

    constructor(props) {
        super(props);
        this.state = {
            data: [],
            page: 0,
            category: ''
        };

        this.SubmitHandler = this.SubmitHandler.bind(this);
        this.returnAllJokes = this.returnAllJokes.bind(this);
        this.getRandom = this.getRandom.bind(this);
    }

    componentDidMount() {

    }

    getResult(){
        this.setState({page : 0, data :[]});
        var searchFor = this.state.text;
        //console.log(searchFor);
        var self = this;
        $('#loading-image').show();
        $.ajax({
            url: "https://api.chucknorris.io/jokes/search?query="+searchFor,
            dataType: 'json',
            success: function(data){
                var filtered = [];
                //console.log(data.result);
                var e = document.getElementById('categories');
                if(e.options[e.selectedIndex].value!='all'){
                    for (var i=0; i<data.result.length; i++){
                        if(data.result[i]['category'] != null) {
                            if(data.result[i]['category'][0] === e.options[e.selectedIndex].value)
                             filtered.push(data.result[i]);
                            //console.log(data.result[i]['category'][0])
                        }
                    }
                }else{
                    filtered = data.result;
                }
                self.setState({data : filtered});
                //console.log(filtered);
            },
            complete: function () {
                $('#loading-image').hide();
            }
        }).done(function(response) {

        });
    }

    SubmitHandler(e) {
        if(e.keyCode===13){
            this.getResult();
        }
        this.setState({text : e.target.value});
    }

    returnAllJokes(){
        var data = this.state.data;
        var self = this;
        if(data.length>0) {
            if (document.getElementById('paginator').checked) {
                return (
                    <div>
                        <p><div style={{'height':'100%'}}><img src={data[0]['icon_url']} alt='chuck'/></div> {data[self.state.page]['value']} </p>
                        <div id='navigator'>
                            {this.state.page>0 ? <button className='form-control' onClick={function () {self.setState({page : self.state.page - 1});}}> PREV </button> : <button disabled className='form-control' onClick={function () {self.setState({page : self.state.page - 1});}}> PREV </button>}
                            <span>{this.state.page+1}/{data.length}</span>
                            {this.state.page<data.length -1 ? <button className='form-control' onClick={function () {self.setState({page : self.state.page + 1});}}> NEXT </button> : <button disabled className='form-control' onClick={function () {self.setState({page : self.state.page + 1});}}> NEXT </button>}
                        </div>
                    </div>
                );
            } else {
                return (
                    data.map(value => {
                        return (<p ><div style={{'height':'100%'}}><img src={value['icon_url']}
                                                                                                      alt='chuck'/></div> {value['value']}</p>);
                    })
                );
            }
        }
    }


    render() {
        return (
            <div className="calc">

                <h2>Chuck Norris API</h2>

                <p>Search for jokes:</p>
                <select className='form-control' id='categories'>
                    {this.getCategories()}
                </select>
                <input placeholder="search" className="form-control" type="text" name="text" onKeyDown={this.SubmitHandler} onChange={this.SubmitHandler}/>
                <button className="btn btn-default" onClick={this.getResult.bind(this)}>Search</button><button className="btn btn-default" onClick={this.getRandom}>Random</button>
                <div><input type='checkbox' className='' onChange={event => {this.setState({page : 0})}} id='paginator'/><label className='' htmlFor="paginator">Enable
                    paginator</label></div>
                <p id="result">
                    <img id='loading-image' hidden src={loader}/>
                {this.returnAllJokes()}
                </p>
            </div>
        );
    }


    getCategories() {
        var self = this;
        var categories = [];
        $.ajax({
            url: "https://api.chucknorris.io/jokes/categories",
            dataType: 'json',
            success: function(data){
                //console.log(data);
                categories = data;
                $('#categories').append($('<option>', {
                    value: 'all',
                    text: 'all'
                }));
                return(
                    categories.map(value => {
                        $('#categories').append($('<option>', {
                            value: value,
                            text: value
                        }));
                    })
                )
            }
        }).done(function(response) {

        });

    }

    getRandom() {
        var self = this;
        var categories = [];
        $.ajax({
            url: "https://api.chucknorris.io/jokes/random",
            dataType: 'json',
            success: function(data){
                //console.log(data);
                var array = [];
                array.push(data);
                self.setState({data : array});
            }
        }).done(function(response) {

        });
    }
}

export default Chuck;
