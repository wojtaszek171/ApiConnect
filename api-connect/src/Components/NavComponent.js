import React, { Component } from 'react';
import {Route, BrowserRouter, NavLink} from "react-router-dom";
import 'jquery-ui-dist/jquery-ui';
import $ from 'jquery';
import Calc from './Calc.js';
import Wiki from './Wiki.js';
import GMap from './Map.js';
import Main from './Main.js'
import './css/Nav.css'

import {LinkContainer} from 'react-router-bootstrap';
import {Navbar, Nav, NavItem} from 'react-bootstrap';

class NavComponent extends Component {

    componentDidMount(){
        $(".nav a").on("click", function(){
            $(".nav").find(".active").removeClass("active");
            $(this).parent().addClass("active");
        });
        $(".navbar-header a").on("click", function(){
            $(".nav").find(".active").removeClass("active");
        });
    }
  render() {
    return (
      <div>
                  <BrowserRouter>
                            <div>
                                <nav className="navbar navbar-default navbar-inverse">
                                    <div className="container-fluid">

                                        <div className="navbar-header">
                                            <button type="button" className="navbar-toggle collapsed"
                                                    data-toggle="collapse" data-target="#bs-example-navbar-collapse-1"
                                                    aria-expanded="false">
                                                <span className="sr-only">Toggle navigation</span>
                                                <span className="icon-bar"></span>
                                                <span className="icon-bar"></span>
                                                <span className="icon-bar"></span>
                                            </button>
                                            <NavLink to="/"><a className="navbar-brand" href="#">API</a></NavLink>
                                        </div>

                                        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                                            <ul className="nav navbar-nav">
                                                <li><NavLink to="/calc"><a href="#">Calcatraz API</a></NavLink></li>
                                                <li><NavLink to="/wiki"><a href="#">Wikipedia API</a></NavLink></li>
                                                <li><NavLink to="/google-map"><a href="#">Google Maps API</a></NavLink></li>
                                            </ul>
                                        </div>
                                    </div>
                                </nav>
                              <div className="content">
                                <Route exact path="/" component={Main}/>
                                <Route path="/calc" component={Calc}/>
                                  <Route path="/wiki" component={Wiki}/>
                                  <Route path="/google-map" component={GMap}/>
                              </div>
                            </div>
                  </BrowserRouter>
      </div>
    );
  }
}

export default NavComponent;
