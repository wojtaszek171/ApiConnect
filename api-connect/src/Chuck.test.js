import React from 'react';
import renderer from 'react-test-renderer';
import Chuck from "./Components/Chuck";
import NavComponent from "./Components/NavComponent";
import { mount, shallow } from 'enzyme';
import ReactTestUtils from 'react-dom/test-utils';
import chuckApi from "./services/chuck";
import serializer from 'enzyme-to-json/serializer';

jest.unmock('jquery');  //jquery unmocked - internet connection needed
var $ = require('jquery') ;


describe('Chuck render',async () => {

    const doneChange = jest.fn();
    const wrapper = shallow(
        <Chuck />);

        var text = wrapper.find("#searchinput");
        const categories = wrapper.find("#categories");
         text.simulate("change", {target:{value:"chucki"}});
         expect(wrapper.state().text).toEqual('chucki');
         //wrapper.find('#search').simulate('click');

});

describe('Chuck api response', async () => {
    const wrapper = shallow(
        <Chuck />);
    jest.setTimeout(20000);
    const datas = await chuckApi('aaa');
    console.log(datas.result.length + " results");
    expect(datas.result.length).toBeGreaterThan(0);
    jest.setTimeout(5000);
    wrapper.setState({data : datas});
    var pag = wrapper.find('#paginator');
    expect(pag.is('[checked]')).toBe(false);
    // pag.simulate('change',{target: {checked: true}});
    // expect(pag.is('[checked]')).toBe(true);
    const p = wrapper.find('#result');
    //expect(p.find(p).exists()).toEqual(true);
});

expect.addSnapshotSerializer(serializer);

describe('Popup', () => {
    it('renders without crashing', () => {
        expect.hasAssertions();
        expect.assertions(2);

        const popup = shallow(<Chuck />);

        expect(popup).toMatchSnapshot();
    });
});