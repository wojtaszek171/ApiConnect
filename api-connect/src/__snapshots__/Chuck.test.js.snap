// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup renders without crashing 1`] = `
<div
  className="calc"
>
  <h2>
    Chuck Norris API
  </h2>
  <p>
    Search for jokes:
  </p>
  <select
    className="form-control"
    id="categories"
    onChange={[Function]}
    value="all"
  />
  <input
    className="form-control"
    id="searchinput"
    name="text"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="search"
    type="text"
  />
  <button
    className="btn btn-default"
    id="search"
    onClick={[Function]}
  >
    Search
  </button>
  <button
    className="btn btn-default"
    onClick={[Function]}
  >
    Random
  </button>
  <div>
    <input
      className=""
      id="paginator"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className=""
      htmlFor="paginator"
    >
      Enable paginator
    </label>
  </div>
  <p
    id="result"
  >
    <img
      hidden={true}
      id="loading-image"
      src="ajaxloader.gif"
    />
  </p>
</div>
`;
